lib/dependabot/update_checkers/java/maven.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require "dependabot/update_checkers/base"
+require "dependabot/git_commit_checker"
+
+module Dependabot
+  module UpdateCheckers
+    module Java
+      class Maven < Dependabot::UpdateCheckers::Base
+        def latest_version
+          # TODO: Hit the registry and get the latest possible version
+        end
+
+        def latest_resolvable_version
+          # TODO: Resolve the pom.xml to find the latest version we could update
+          # to without updating any other dependencies at the same time
```

Comment:
This is the hard bit. Finding the latest version of a dependency is normally pretty easy. Finding the latest version that is compatible with the rest of your project's dependencies is tricky. Generally, we see two options:
1) Some languages don't do resolution at all (e.g., Python and JavaScript). In that case this step is trivial, and we just return the latest version from the registry
2) Most languages *do* do resolution, in which case we shell out to a helper that is written in that language and piggy-back off of the resolver's code to figure out what the latest compatible version is *without* actually doing a full install (which would take ages)

Reply:
I'm not very familiar with how Maven works internally, but I suppose something like the [Apache Maven Dependency plugin](https://maven.apache.org/plugins/maven-dependency-plugin/index.html) could be used. I just came across [this plugin](http://www.mojohaus.org/versions-maven-plugin/index.html) as well.